cfpb/cfgov-refresh
# Add owning-a-home/explore-rates scripts and dependencies

Apps (such as owning-a-home) may have their own esoteric dependencies that are not applicable to other aspects of consumerfinance.gov. It therefore makes sense to provide these projects with their own package.json to manage their own dependencies, which can be pulled into the webpack pipeline for that app's scripts.

## Additions

- Moves /explore-rates/ owning-a-home files into the project and adds dependencies.

## Changes

- Adds rate-checker dependencies to oah's package.json.

## Testing

1. Go into `urls.py` and change:
```
url(r'^owning-a-home/explore-rates/',
        include(oah.urls_for_prefix('explore-rates'))),
```

To:

```
url(r'^owning-a-home/explore-rates/',
        TemplateView.as_view(
        template_name='owning-a-home/explore-rates/index.html'),
        name='explore-rates'),
```

2. Restart the server.
3. Run `gulp clean`.
4. From project root, run `cp ./cfgov/unprocessed/apps/owning-a-home/example-config.json ./cfgov/unprocessed/apps/owning-a-home/config.json`.
5. Run `gulp scripts:apps`.
6. Verify message appears: `App dependencies not installed, please run from project root: npm --prefix ./cfgov/unprocessed/apps/owning-a-home install ./cfgov/unprocessed/apps/owning-a-home`.
7. Run  `npm --prefix ./cfgov/unprocessed/apps/owning-a-home install ./cfgov/unprocessed/apps/owning-a-home` from project root.
8. Re-run `gulp scripts:apps` and verify it passes.
9. Run `gulp build`.
10. Check that http://localhost:8000/owning-a-home/explore-rates/ has no console error. (Note: layout will not be correct, but this PR is just about getting the JS assets into that page without errors.)

## Notes
- Contains https://github.com/cfpb/cfgov-refresh/pull/3817
- Contains https://github.com/cfpb/cfgov-refresh/pull/3819
- Contains https://github.com/cfpb/cfgov-refresh/pull/3820
